cumulusci/cumulusci.yml

```diff
@@ -464,13 +464,21 @@ tasks:
         description: Uploads a beta release of the metadata currently in the packaging org
         class_path: cumulusci.tasks.salesforce.PackageUpload
         group: Release Management
+    upload_default_user_profile_photo:
+        group: Data Operations
+        description: Uploads a profile photo for the default User.
+        class_path: cumulusci.tasks.salesforce.UploadDefaultUserProfilePhoto
     upload_production:
         description: Uploads a production release of the metadata currently in the packaging org
         class_path: cumulusci.tasks.salesforce.PackageUpload
         options:
             name: Release
             production: True
         group: Release Management
+    upload_user_profile_photo:
+        group: Data Operations
+        description: Uploads a profile photo for the User whose user_field equals user_field_value.
+        class_path: cumulusci.tasks.salesforce.UploadUserProfilePhoto
```

Comment:
Yes.   I've been weighing that question, and I like the key/value option.  I could even take a list of key/value pairs which translate to a list of `AND` filters for the User SOQL query.

Reply:
Hmm, maybe just better to take the entire SOQL WHERE clause as an string option instead of trying to invent a one-off, incomplete way of building SOQL expressions?

Reply:
I have an idea that I think works well.   I check the field's [SOAP Type](https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_enum_Schema_SOAPType.htm).   If the SOAP Type "text" meaning STRING (`xsd:string`), ID (`tns:ID`), or ADDRESS (`urn:address`), I wrap the field value in single quotes.  Else, don't wrap the field in single quotes. 

It seems pretty effective so the query supports Booleans and even date/datetime literals.

Reply:
Accepting the whole WHERE clause as a string would also support those things, as well as supporting everything else that's valid in a WHERE clause (OR, IN, LIKE, comparison operators, checks for equality to null...).

I'm not asking you to implement support for those things. I'm sure you could. But I don't think generic logic for building a SOQL expression from a Python structure belongs in this task, and I don't see a particular reason we need it.

Any thoughts, @davidmreed?

Reply:
I agree it's more generic, and I will update to support a `filter` SOQL rather than `filters`.    My heart liked how the implementations would look.   

```yml
tasks:
    dev_upload_profile_photo_grace:
        group: "SFDO-Grants: Internal config and storytelling"
        description: Uploads a profile photo for (business account) grantseeker Grace Walker.
        class_path: cumulusci.tasks.salesforce.users.UploadProfilePhoto
        options:
            photo: dev/photos/grace.png
            filters:
                FirstName: Grace
                LastName: Walker
```

Reply:
+1 `where` option give the most flexibility plus it seems simpler to implement

Reply:
I agree with @davisagli. The WHERE clause is a well-established CCI pattern that fits here. Supporting a kind of SOQL builder in the options has merit, but I think we should evaluate that option as part of a bigger rethink of how options work across tasks rather than introduce a one-off mechanic that will need to be maintained as backwards-compatible.

Reply:
implemented and pushed.